### DISCUSSION TITLE
Feature: Voting up issues

### DISCUSSION THREAD

**Discussion by stuymedova** [community/community#17119] [Projects and Issues]
I think it would be good if users were able to vote up issues that raise relevant to them problems. As such, maintainers would know which problems need more attention, and would know to prioritize them. I see this being most useful for large projects with thousands of users and hundreds of contributors (and issues).

**Comment by svick** [comment]
There is already a way to do that: use the 👍 reaction. Discussions have both emoji reactions and upvotes and I find it quite confusing: how are you supposed to interpret e.g. 5 ⬆️ 7 👍 ?

**Reply by stuymedova** [reply]
Sorting by 👍 reaction seems to be a workaround to achieve the effect of sorting by upvotes. Even if people may be accustomed to it currently, I believe a clear and intuitive feature such as upvoting would be used primary, even if there'd be a transition period.

That said, I hope sorting by upvotes to replace sorting by emojis, as I believe the latter is suboptimal. An issue receiving 100 upvotes and being the most upvoted issue would send a more clear message, than it receiving 25 👍 emojis, 25 😄 emojis, 25 👀 emojis, and 25 🚀 emojis, and not being on top of either list.

I also believe that a number of people upvoting would be higher than the number of people reacting with emojis. If I see an issue that is important to me, it's more intuitive for me to upvote it, not look for an emoji to react with. (In fact, it seems to be true for discussions).

And now, when emojis aren't used for sorting anymore, there could be added a wider selection of emojis to react with. Perhaps 🔥 is someone's most preferable emoji.

**Reply by sherryumlah** [reply]
It doesn't appear that you can sort on `sort:reactions-+1` for the github projects view, though.  
<img width="392" alt="Screen Shot 2022-05-31 at 11 54 28 AM" src="https://user-images.githubusercontent.com/11161280/171230600-26b1a6b9-33f5-4129-ad63-ce6903f17c39.png">

**Reply by samzuercher** [reply]
I also think something similar to stackoverflow would be cool for github to solve the problem not only with **Issues**, but also with **Discussions** and probably more types in GitHub: 

## which of all the answers and questions are relevant or solved the problem?

Currently we have to scroll down and search for it, probably sort on something.. It looks a lot easier on stackoverflow: you can simply upvote and everyone gets the top answers straight away without any filtering nor scrolling and searching.

**Comment by Gouvernathor** [comment]
I suggest something else than the previous answer : having a boolean upvote/no-upvote button for each issue for each user (including the maintainers), or possibly a ternary upvote/novote/downvote button like stackoverflow, ***and*** having the issues sorted (by default) by decreasing number of maintainer upvotes, then by decreasing number of total upvotes.
That, and an opt-out (or an opt-in) for maintainers not wanting the public to have upvote/downvote power.

**Comment by ghost** [comment]
Hi all! this is... duplicate of: https://github.com/github-community/community/discussions/18483

**Reply by LucBerge** [reply]
No. It is the opposite. 17119 < 18483

**Comment by rickmark** [comment]
More data:

Most people dislike the "+1" method of voting, even reactmojiis are pretty useless.

There should be two elements to this "feature"

A prominent (above the fold) listing of count of votes and ability to add one, possibly similar to the way a branch becomes obvious for a new PR on the code tab.

The ability of the UI to detect a comment that contains no "content" and redirect that into a vote call-to-action or a vote themselves.

This all seems pretty obvi since the GitHub community feature requests has many of the correct features already.  Also the possibility to "merge in" the duplicate issue comments at the right point to keep any useful discussion.

**Comment by sypets** [comment]
I think that a clear :arrow_up:  upvote  / downvote :arrow_down:  button (or a `+1` and` -1`)  is better than a :thumbsup:  / :thumbsdown: .

The current thumbsup / thumbsdown for GitHub issues (or PR) is insufficient to use as "voting feature" for a number of reasons:

* you don't see the :thumbsup:, :thumbsdown: in the issues list by default, only if you sort by emoticon
* If you sort by emoticon, you can only choose one, e.g. +1 **or** -1 . So for very controversial issues with lots of "up" **and** "downvotes" (via thumbsup / thumbsdown) this is misleading
* There is no way to tally (create a sum) and sort by this, e.g. ` sum = count_upvotes + count_downvotes * -1`
* The various emoticons (with :heart:, rocket etc.) are not helpful for tallying a "community approval"
* Weighing maintainer votes more heavily would be quite helpful. This is currently not possible.

In addition, I agree with the sugestion of [Gouvernathor](https://github.com/Gouvernathor) but I think a **downvote** should **definitely** be an option (there are plenty of good reasons for downvoting things to oblivion, security concerns is only one of them):

> I suggest something else than the previous answer : having a boolean upvote/no-upvote button for each issue for each user (including the maintainers), or possibly a ternary upvote/novote/downvote button like stackoverflow, and having the issues sorted (by default) by decreasing number of maintainer upvotes, then by decreasing number of total upvotes.
That, and an opt-out (or an opt-in) for maintainers not wanting the public to have upvote/downvote power.

https://github.com/orgs/community/discussions/17119#discussioncomment-2974587

**Comment by rolfsmeds** [comment]
👍 

(yes, this was an intentionally wrong way to upvote this proposal to demonstrate the problem with emojis – many users add their reaction emojis as comments instead of reactions to the top-comment)

**Comment by leagris** [comment]
Microsoft, the ball is in your court now!
